agent/config/config.go

```diff
@@ -872,12 +877,28 @@ type TLSProtocolConfig struct {
 	UseAutoCert          *bool   `mapstructure:"use_auto_cert"`
 }
 
+func (c TLSProtocolConfig) IsZero() bool {
+	v := reflect.ValueOf(c)
+
+	hasValues := false
+	for i := 0; i < v.NumField(); i++ {
+		if !v.Field(i).IsNil() {
+			hasValues = true
+		}
+	}
+	return !hasValues
+}
+
 type TLS struct {
 	Defaults    TLSProtocolConfig `mapstructure:"defaults"`
 	InternalRPC TLSProtocolConfig `mapstructure:"internal_rpc"`
 	HTTPS       TLSProtocolConfig `mapstructure:"https"`
 	GRPC        TLSProtocolConfig `mapstructure:"grpc"`
 
+	// SpecifiedTLSStanza indicates whether the per-protocol tls stanza from configuration was used.
+	// If unspecified, and TLS is configured, that implies that the deprecated flags were used.
+	SpecifiedTLSStanza *bool
```

Comment:
do we need `mapstructure:"-"` here? I'm assuming this is not meant to be user-defined